src/utils/proxy.ts

```diff
@@ -757,6 +757,28 @@ const onRequest = async (
     return proxy.web(req, res, { target: edgeFunctionsProxyURL })
   }
 
+  const match = await rewriter(req)
+  const options = {
+    match,
+    addonsUrls,
+    target: `http://${
+      settings.frameworkHost && isIPv6(settings.frameworkHost) ? `[${settings.frameworkHost}]` : settings.frameworkHost
+    }:${settings.frameworkPort}`,
+    publicFolder: settings.dist,
+    functionsServer,
+    functionsPort: settings.functionsPort,
+    jwtRolePath: settings.jwtRolePath,
+    framework: settings.framework,
+  }
+
+  if (match) {
```

Comment:
what if the match is not a redirect, but a rewrite? we'll want to match functions on the rewritten path in that case